guides/puppet-web-apps/html-views/README.md

```diff
@@ -0,0 +1,98 @@
+# HTML Views
+
+This page gives an explaination of partial HTML views and view-models, or, in other words, nested pages.
+
+## Partial HTML View
+
+A Partial HTML View is a valid HTML document that contains at least one `<template>` tag and can be loaded with the [`<imported-template>`](https://github.com/Juicy/imported-template) web component. The `<template>` tag is cached on the client and stamped out as many times as needed.
+
+The requirements and behaviors for partial HTML views originate from the [HTML Imports](http://www.w3.org/TR/html-imports/) and [HTML Template](http://www.w3.org/TR/html-templates/) specifications. The most interesting aspects are:
+
+* The document should be a valid HTML document and can contain anything that the browser allows
+* The document may include dependencies such as stylesheets, HTML imports, scripts, etc.
+* The content outside of the `<template>` tag is requested and executed once per session on partial load time
+* The content inside the `<template>` is stamped out with the `<template>` itself every time the response is bound to a tree in the view-model
+* Every node from the `<template>` will be stamped with an attached model property. Custom elements, template binding frameworks, or JavaScript scripts can use this data to populate the view
+
+## Partial View-Model
+
+A partial view-model is a valid JSON document with the filename extension `.json` that contains an `Html` property that points to the corresponding partial HTML view. It will also contain properties for the data that will be bound between the view and the database model.
+
+The partial view-model can be combined with an optional view-model code-behind which defines the class name and contains input handlers. The filename extension for this file is `.json.cs`.
+
+A partial HTML view together with its corresponding partial view-model can simply be referred to as a partial.
+
+### Using Partials
+
+A partial can be accessed in two different ways:
+
+1. Directly through an HTTP request that is handled in the application by a `Handle.GET`. The response to this request should be a complete HTML and JSON document. 
+2. Through an **insertion point**. An insertion point is a combination of a `Self.GET` call, which acts as the partial view-model insertion point, and a [`<starcounter-include>`](https://github.com/Polyjuice/starcounter-include) tag, which acts as a partial HTML view insertion point. The `<starcounter-include>` tag determines where in the DOM tree the partial HTML view should be rendered.
```

Comment:
The URL https://github.com/Polyjuice/starcounter-include should be changed to https://github.com/Starcounter/starcounter-include